pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -561,7 +560,12 @@ func convertResolution(resolution model.Resolution) apiv2.Cluster_DiscoveryType
 }
 
 // conditionallyConvertToIstioMtls fills key cert fields for all TLSSettings when the mode is `ISTIO_MUTUAL`.
-func conditionallyConvertToIstioMtls(tls *networking.TLSSettings, serviceAccounts []string, sni string) *networking.TLSSettings {
+func conditionallyConvertToIstioMtls(
+	tls *networking.TLSSettings,
+	serviceAccounts []string,
+	sni string,
+	metadata map[string]string,
```

Comment:
nit: please pass proxy here instead of proxy.metadata so that it becomes easy to spot code that relies on potentially proxy specific info

Reply:
fixed